src/mds/Server.cc

```diff
@@ -501,6 +507,16 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
     return;
   }
 
+  fs_cluster_id_t fsid = mds->get_fsid();
+  if (fsid != FS_CLUSTER_ID_NONE && !session->fsid_capable(fsid, MAY_READ)) {
+    dout(0) << " dropping message not allowed for this fsid: " << *m << dendl;
+    auto reply = make_message<MClientSession>(CEPH_SESSION_REJECT);
+    reply->metadata["error_string"] = "client doesn't have caps for fsid " +
```

Comment:
remove this line. old kernels treat session message with metadata as bad message

Reply:
Oh, okay.